packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -136,6 +139,19 @@ export class RoutesMatcher {
 			return;
 		}
 
+		// If the new destination is an intercept route, only allow it if the current path is also an
+		// intercept route. The build output config correctly maps relevant request paths to be
+		// intercepts in the `none` phase, while the `rewrite` phase can contain entries that rewrite
+		// to an intercept that matches requests that are not actually intercepts, causing a 404.
```

Comment:
Here to me mentioning the phases is a bit confusing as they don't seem to be relevant, I would suggest to reduce this comment
```suggestion
		// If the new destination is an intercept route, only allow it if the current path is also an
		// intercept route.
```
and add the rest below above where you set the `checkIntercept`:
```diff
+       // The build output config correctly maps relevant request paths to be
+		// intercepts in the `none` phase, while the `rewrite` phase can contain entries that rewrite
+		// to an intercept that matches requests that are not actually intercepts, causing a 404.
        checkIntercept: phase === 'rewrite',
```

What do you think? 🙂